### ISSUE TITLE
delayed rule queue should be sorted by exe_time

### ISSUE THREAD

**Issue by tedgin** [irods/irods#3781]
I'm using iRODS 4.1.10.

We used delayed rules to automate a few of our processes.  Replicating newly uploaded files to a resource server in a remote data center is one. Many of our users uploaded large sets of files at a time. A large batch of the resulting replication rules scheduled at the same time would interrupt our other processes making our infrastructure less responsive for some of our more interactive applications.  To reduce the impact of the large set uploads, each of resulting rules is scheduled further and further in the future. This allows rules generated by the actions of other users to executed in between the execution of these delayed replication rules.

When monitoring delayed queue,  I noticed that when the queue grew long, the rodsReServer workers seemed to prefer rules scheduled to run close to the current time over rules scheduled to run further in the past.  Looking at the SQL query being executed, the workers appear to prefer the execute the rule with the oldest creation time that has an execution time that isn't in the future.  

As a consequence of this rule selection logic, a delayed rule created 10 hours ago to be executed in 10 hours (now) would be chosen over a delayed rule created 9 hours ago to be executed in 1 hour (8 hours ago).  That seems backwards to me.

**Comment by trel** [comment]
Seems reasonable... @xu-hao can you share any insight for this ordering?

Link to the SQL?

**Comment by tedgin** [comment]
I just looked at the query log on the DBMS when to see the SQL.

```sql
select distinct R_RULE_EXEC.rule_exec_id ,R_RULE_EXEC.rule_name ,R_RULE_EXEC.rei_file_path ,R_RULE_EXEC.user_name ,R_RULE_EXEC.exe_address ,R_RULE_EXEC.exe_time ,R_RULE_EXEC.exe_frequency ,R_RULE_EXEC.priority ,R_RULE_EXEC.estimated_exe_time ,R_RULE_EXEC.notification_addr ,R_RULE_EXEC.last_exe_time ,R_RULE_EXEC.exe_status  from  R_RULE_EXEC  order by R_RULE_EXEC.rule_exec_id
```

**Comment by trel** [comment]
Okay, so it seems updating the `order by` part of that query would perform the queued jobs in the order you expect:
```sql
order by R_RULE_EXEC.exe_time
```

**Comment by xu-hao** [comment]
Yes, that should work. Tony, thanks for suggesting the SQL statement.

**Comment by xu-hao** [comment]
Upon further reading, I think this is behavior change. I think the current
design may have use cases. From technical point of view it seems the change
would work.
